src/util/pup_util.C

```diff
@@ -649,33 +649,34 @@ void PUP::toTextUtil::endLine(void) {
 void PUP::toTextUtil::beginEnv(const char *type,int n)
 {
   char *o=beginLine();
-  sprintf(o,"begin "); o+=strlen(o);
-  sprintf(o,type,n); o+=strlen(o);
-  sprintf(o," {\n");
+  snprintf(o,maxCount,"begin "); o+=strlen(o);
+  snprintf(o,maxCount-strlen(o),type,n); o+=strlen(o);
+  snprintf(o,maxCount-strlen(o)," {\n");
```

Comment:
These `strlen`s will not give the intended result because `o` is modified to point to the end of the written contents, unlike the place in controlPoints.h which passes in a temporary value of `s+slen` each time.

For minimal disruption to the code, the easiest approach would probably be replacing the variable definition with:

```c++
char * const begin=beginLine();
char *o=begin;
```

and replace `strlen(o)` in the size parameter with `(o-begin)`, parentheses important.